ci-operator/templates/build-azure-image-310.yaml

```diff
@@ -0,0 +1,317 @@
+          -e openshift_azure_storage_account_ns=images \
+          -e openshift_azure_container=images \
+          -e openshift_azure_vm_size=$AZURE_VM_SIZE \
+          -e openshift_azure_install_repo=$(curl -qs https://storage.googleapis.com/origin-ci-test/releases/openshift/origin/release-${OPENSHIFT_RELEASE}/.latest-rpms) \
+          playbooks/azure/openshift-cluster/build_node_image.yml
+
+        az login --service-principal -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} --tenant ${AZURE_TENANT} &>/dev/null
+        az group delete -n ${RESOURCE_GROUP} -y
+
+        touch /tmp/shared/image_create
+
+    # Runs an install
+    - name: buildcluster
+      image: ${LOCAL_IMAGE_BIN}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-secrets-azure
+        mountPath: /etc/azure/credentials
+      env:
+      - name: TYPE
+        value: ${CLUSTER_TYPE}
+      - name: HOME
+        value: /tmp/shared/home
+      - name: OPENSHIFT_RELEASE
+        value: "${OPENSHIFT_RELEASE}"
+      - name: DEPLOY_OS
+        value: "${DEPLOY_OS}"
+      command:
+      - /bin/bash
+      - -c
+      - |
+       #!/bin/bash
+        # error handling and sync code
+        set -euo pipefail
+        echo "starting buildcluster"
+
+        trap '[ "$?" == 0 ] || touch /tmp/shared/exit' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        # Cluster creation specific configuration.
+        mkdir -p "${HOME}"
+
+        # wait until the setup job creates admin.kubeconfig
+        while true; do
+          if [[ ! -f  /tmp/shared/image_create ]]; then
+            echo "sleeping waiting for image_create"
+            sleep 15 & wait
+            continue
+          fi
+          # if we got admin kubeconfig file with failure marker, ignore. Teardown is in progress.
+          if [[ -f  /tmp/shared/exit ]]; then
+             echo "Exiting.  Found /tmp/shared/exit."
+             exit 1
+          fi
+          break
+        done
+        echo "Found image create marker file."
+
+        source /etc/azure/credentials/secret
+        az login --service-principal -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} --tenant ${AZURE_TENANT_ID} &>/dev/null
+        # TODO: Configure these from prow
+        export DNS_DOMAIN=osadev.cloud
+        export DNS_RESOURCEGROUP=dns
+        export DEPLOY_VERSION=v${OPENSHIFT_RELEASE}
+        export RUN_SYNC_LOCAL=true
+        export IMAGE_RESOURCEGROUP=images
+
+        source /tmp/shared/environment
+        export IMAGE_RESOURCENAME=${IMAGE_NAME}
+        cd /go/src/github.com/openshift/openshift-azure/
+
+        # added timeout as this seems to loop infinitely awaiting cluster deployment
+        timeout 30m ./hack/create.sh ${RESOURCE_GROUP}-e2e
+
+        # if completed pass generated data to other stage
+        cp -r /go/src/github.com/openshift/openshift-azure/_data /tmp/shared/
+
+        touch /tmp/shared/e2e_tests
+
+    - name: tagimage
```

Comment:
Is there any reason for a separate container for doing the tagging? It seems we would reduce the size of bash we maintain by moving this after running the node build playbook?

Reply:
Sorry, just realized you need to tag after the e2e test succeeds. Nvmd.